pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -145,6 +148,16 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable SqlNodeAndOption
       }
     }
 
+    // check app qps before doing anything
+    String application = sqlNodeAndOptions.getOptions().get(Broker.Request.QueryOptionKey.APPLICATION_NAME);
+    if (application != null && !_queryQuotaManager.acquireApplication(application)) {
+      String errorMessage =
+          String.format("Request %d: %s exceeds query quota for application: %s", requestId, query, application);
```

Comment:
Why using format here? Isn't it just easier, easier to read (and cheaper!) to concatenate?

Reply:
This follows style of existing code, e.g. MultiStageBrokerRequestHandler:216. 
String + would be equally bad and StringBuilder not that much better to be honest.

Reply:
I find quite easier to read a string built with `+` than String.format or StringBuilder. And in terms of performance String.format is orders of magnitude worse given it needs to parse the string, find all the special arguments like %d or %s. Then it needs to process them (understand that %d and %s are different) and then it needs to allocate an Object[] to store requestId, query and application (which hopefully should be optimized by JIT, but we cannot be sure).

It is not that in this case performance actually matters, but unless we need to apply some custom format (ie using exponential annotation) it is never worth to use String.format. Even if it is used somewhere else, we should reduce it usage instead of extending it.

Reply:
I don't like String.format() either but to me it creates garbage of quantity comparable with String concat.
I only used it to follow the style of db qps code. I'll change it here but reckon what we ought to do is remove the  worse pattern of 'String.format()' from the code base altogether. This way any usage will stick out like a sore thumb.